client/reader/feed-stream/index.jsx

```diff
@@ -1,94 +1,109 @@
+		if ( _feed ) {
+			_feed.site_icon = getReaderFollowForFeed( state, parseInt( feedId ) )?.site_icon;
 		}
 
-		return (
-			<Stream
-				{ ...this.props }
-				listName={ title }
-				emptyContent={ emptyContent }
-				showPostHeader={ false }
-				showSiteNameOnCards={ false }
-			>
-				<DocumentHead
-					title={ this.props.translate( '%s ‹ Reader', {
-						args: title,
-						comment: '%s is the section name. For example: "My Likes"',
-					} ) }
-				/>
-				<ReaderFeedHeader
-					feed={ feed }
-					site={ site }
-					showBack={ this.props.showBack }
-					streamKey={ this.props.streamKey }
-				/>
-				{ ! feed && <QueryReaderFeed feedId={ this.props.feedId } /> }
-				{ siteId && <QueryReaderSite siteId={ siteId } /> }
-			</Stream>
-		);
-	}
-}
+		return _feed;
+	} );
+
+	const { isBlocked, postCount, site, siteId } = useSelector( ( state ) => {
+		const _siteId = getReaderSiteId( feed );
 
-export default connect( ( state, ownProps ) => {
-	const feed = getFeed( state, ownProps.feedId );
-	const siteId = getReaderSiteId( feed );
+		return {
+			isBlocked: _siteId && isSiteBlocked( state, _siteId ),
+			postCount: _siteId && getAllPostCount( state, _siteId, 'post', 'publish' ),
+			site: _siteId && getSite( state, _siteId ),
+			siteId: _siteId,
+		};
+	} );
 
-	// Add site icon to feed object so have icon for external feeds
-	if ( feed ) {
-		const follow = getReaderFollowForFeed( state, parseInt( ownProps.feedId ) );
-		feed.site_icon = follow?.site_icon;
+	const siteTags = useSiteTags( siteId );
+	const emptyContent = <EmptyContent />;
+	const title = getSiteName( { feed, site } ) || translate( 'Loading Feed' );
+	const followerCount = getFollowerCount( feed, site );
+
+	if ( isBlocked ) {
+		return <SiteBlocked title={ title } siteId={ siteId } />;
+	}
+
+	if ( ( feed && feed.is_error ) || ( site && site.is_error ) ) {
+		return <FeedError sidebarTitle={ title } />;
 	}
 
-	return {
-		feed,
-		siteId,
-		site: siteId && getSite( state, siteId ),
-		isBlocked: isSiteBlocked( state, siteId ),
-	};
-} )( localize( FeedStream ) );
+	const streamSidebar = (
+		<FeedStreamSidebar
+			followerCount={ followerCount }
+			postCount={ postCount }
+			tags={ siteTags.data }
+		/>
+	);
+
+	return (
+		<Stream
+			{ ...props }
+			className={ className }
+			listName={ title }
+			emptyContent={ emptyContent }
+			showPostHeader={ false }
+			showSiteNameOnCards={ false }
+			streamSidebar={ streamSidebar }
+		>
+			<DocumentHead
+				title={ translate( '%s ‹ Reader', {
+					args: title,
+					comment: '%s is the section name. For example: "My Likes"',
+				} ) }
+			/>
+			<ReaderFeedHeader
+				feed={ feed }
+				site={ site }
+				showBack={ showBack }
+				streamKey={ props.streamKey }
+			/>
+			{ ! feed && <QueryReaderFeed feedId={ feedId } /> }
+			{ siteId && <QueryReaderSite siteId={ siteId } /> }
+			{ siteId && <QueryPostCounts siteId={ siteId } type="post" /> }
+		</Stream>
+	);
+}
```

Comment:
Note that most of the changes in this file are to convert the FeedStream into a functional component, which is encapsulated in https://github.com/Automattic/wp-calypso/pull/78702--I'll plan to merge that PR first.